agent/lib/src/health.dart

```diff
@@ -131,3 +139,24 @@ Future<HealthCheckResult> _scrapeRemoteAccessInfo() async {
       ? 'Unable to determine IP address. Is wired ethernet connected?'
       : 'Last known IP address: $ip');
 }
+
+/// Completely removes Xcode DerivedData directory.
+///
+/// There're two purposes. First, it's a well known trick to fix Xcode when
+/// Xcode behaves strangely for no obvious reason. Second, it avoids eating
+/// all of the remaining disk space over time.
+@visibleForTesting
+Future<HealthCheckResult> removeXcodeDerivedData(
+    {platform.Platform pf = const platform.LocalPlatform(),
+    file.FileSystem fs = const local.LocalFileSystem()}) async {
+  if (!pf.isMacOS) {
+    return HealthCheckResult.success();
+  }
+  String home = pf.environment['HOME'];
+  if (home == null) {
+    return HealthCheckResult.failure('Missing \$HOME environment variable.');
+  }
+  String p = path.join(home, 'Library/Developer/Xcode/DerivedData');
+  rrm(fs.directory(p));
```

Comment:
Actually will this crash if the directory doesn't exist?  We wouldn't want two heath checks in a row to cause a crash.

Reply:
The `rrm` function somehow doesn't crash in this case, I have tested manually and in a unit test.